### ISSUE TITLE
Support for Logstash Beats plugin (Filebeat)

### ISSUE THREAD

**Issue by richard-keller** [whyscream/postfix-grok-patterns#79]
The current config assumes use of Logstash's syslog input to receive postfix logs. However, this implementation won't work if logs are transported using the [Logstash Beats plugin](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-beats.html) instead of the syslog input, since the `program` field will not be present.

This should be mentioned in the docs. The following fix works for me. For people using Logstash Beats, by adding this section before the very first `if` statement in the config file, the `program` field will be available to subsequent expressions. Note that this assumes log lines to be in default syslog format.

```
    grok {
        match => { "message" => "%{SYSLOGTIMESTAMP} %{SYSLOGHOST} %{DATA:program}(?:\[%{POSINT}\])?: %{GREEDYDATA}" }
    }
```

**Comment by hacktvist** [comment]
@richard-keller did you saw any performance improvements after using filebeat as a transporter.

**Comment by richard-keller** [comment]
@hacktvist  To be honest, I didn't compare. This is the first time I've set up Logstash, and I only ended up using filebeat because it appears to be one of the preferred transporters.

**Comment by whyscream** [comment]
I'll expand the README with a list of required input fields, and some suggestions on fixing input from other plugins.

**Comment by whyscream** [comment]
Solved in #81

**Comment by sastorsl** [comment]
Since I have many sources I wrapped *logstash.conf in an "if postfix" and added grok immediately after.
